datafusion/physical-expr/src/functions.rs

```diff
@@ -2843,16 +2844,16 @@ mod tests {
 
             match expr {
                 Ok(..) => {
-                    return Err(DataFusionError::Plan(format!(
+                    return plan_err!(
                         "Builtin scalar function {fun} does not support empty arguments"
-                    )));
+                    );
                 }
                 Err(DataFusionError::Plan(err)) => {
                     if !err
                         .contains("No function matches the given name and argument types")
                     {
-                        return Err(DataFusionError::Internal(format!(
-                            "Builtin scalar function {fun} didn't got the right error message with empty arguments")));
+                        return plan_err!(
```

Comment:
Does this change from Internal error to Plan error?

Reply:
it handles the condition related to 
```
Err(DataFusionError::Plan(err))
```
thus its likely more related to plan than to Internal